### PR TITLE
legacy: addition of feature flag to disalbe redirect to legacy

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

File: backend/inspirehep/legacy/views.py [hunk 1]
```diff
@@ -46,54 +50,80 @@ def _redirect_author(bai):
...
-        302,
+    if current_app.config["FEATURE_FLAG_ENABLE_LEGACY_VIEW_REDIRECTS"]:
+        return redirect(
+            f"{current_app.config['LEGACY_BASE_URL']}/search?{request.query_string.decode('utf-8')}",
+            302,
+        )
+
+    collection = MAP_LEGACY_COLLECTION_TO_HIDDEN_COLLECTION.get(
+        legacy_collection.lower()
     )
```
Comment (michamos) [inline backend/inspirehep/legacy/views.py:100]: sorry, maybe I wasn't clear in zulip. I don't believe this is needed, as we've already aligned the metadata in `_collections` with the legacy collection name in dojson. So you can do (case normalization is already handled by ES, so not needed either).
```suggestion
    collection = legacy_collection
```

File: backend/inspirehep/legacy/views.py [hunk 2]
```diff
@@ -46,54 +50,80 @@ def _redirect_author(bai):
...
+    if collection:
+        query = f" and {query}" if query else ""
+        query = f"_collection:{collection}{query}"
```
Comment (michamos) [inline backend/inspirehep/legacy/views.py:103]:
```suggestion
        query = f'_collections:"{collection}"{query}'
```

File: backend/inspirehep/legacy/views.py [hunk 3]
```diff
@@ -46,54 +50,80 @@ def _redirect_author(bai):
...
-        f"{current_app.config['LEGACY_BASE_URL']}/collection/{legacy_collection}", 302
+    if current_app.config["FEATURE_FLAG_ENABLE_LEGACY_VIEW_REDIRECTS"]:
+        return redirect(
+            f"{current_app.config['LEGACY_BASE_URL']}/collection/{legacy_collection}",
+            302,
+        )
+
+    collection = MAP_LEGACY_COLLECTION_TO_HIDDEN_COLLECTION.get(
+        legacy_collection.lower()
     )
```
Comment (michamos) [inline backend/inspirehep/legacy/views.py:122]: same story here
```suggestion
    collection = legacy_collection
```

File: backend/inspirehep/legacy/views.py [hunk 4]
```diff
@@ -46,54 +50,80 @@ def _redirect_author(bai):
...
+    if collection:
+        query = f"_collection:{collection}"
```
Comment (michamos) [inline backend/inspirehep/legacy/views.py:124]:
```suggestion
        query = f"_collections:{collection}"
```

**Comment by drjova** [comment]
@michamos updated :)

**Review by michamos — APPROVED** [review]

**Pull request merged**